packages/blocks/src/__internal__/utils/types.ts

```diff
@@ -241,3 +248,24 @@ export type Detail<T extends keyof WindowEventMap | keyof HTMLElementEventMap> =
     : T extends keyof HTMLElementEventMap
     ? HTMLElementEventDetail<T>
     : never;
+
+export abstract class AbstractSelectionManager<
```

Comment:
@Saul-Mirone how do you think about adding this abstraction? Seems the having the DOM `container` for host makes sense to us.

Reply:
I think the selection manager won't be a long term pattern. I'll redesign it when working on the plugin spec.